include/ttmlir/Dialect/TTIR/IR/TTIROps.td

```diff
@@ -435,6 +435,38 @@ def TTIR_Conv2dOp : TTIR_DPSOp<"conv2d"> {
     let hasVerifier = 1;
 }
 
+def TTIR_MaxPool2dOp : TTIR_DPSOp<"max_pool2d"> {
+    let summary = "Applies a 2D max pooling over an input signal composed of several input planes.";
+    let description = [{
+      Applies a 2D max pooling over an input signal composed of several input planes.
+    }];
+
+    let arguments = (ins AnyRankedTensor:$input,
+                         AnyRankedTensor:$output,
+                         SI32Attr:$input_height,
+                         SI32Attr:$input_width,
+                         SI32Attr:$kernel_height,
+                         SI32Attr:$kernel_width,
```

Comment:
Btw, all of these can be changed from:
```
SI32Attr:$stride_height,
```

To:
```
"int64_t":$stride_height,
```

And then tablgen will generate the getters and setters to work with ints instead of IntAttr's, which is easier to work with.

Reply:
@nsmithtt Since pooling doesn't have weights there is nothing to infer the kernel size from. Furthermore the TTNN maxpool2d op requires the input be `(1, 1, N*H*W, C)`, so the height width and batch size information is lost by the time we get to runtime, unless they are stored as attributes.

Reply:
> TTNN maxpool2d op requires the input be (1, 1, N*H*W, C)

Yes the op should be defined this way in TTNN dialect, but in TTIR we shouldn't define it this way.  Agree about the kernel size, but input_height/width should be removed.